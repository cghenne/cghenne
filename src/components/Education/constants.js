export const education = [
  {
    graduation_year: 2008,
    school: 'Paul Lambin Institute',
    location: 'Brussels, Belgium',
    description: `BSc Computer Science (web services, infographic
      and artificial intelligence)`
  },
  {
    graduation_year: 2005,
    school: 'High School "Lyc√©e de Berlaymont"',
    location: 'Waterloo, Belgium',
    description: `Sciences and mathematics`
  }
];

export type SchoolT = {
  graduation_year: number,
  school: string,
  location: string,
  description: string
};
