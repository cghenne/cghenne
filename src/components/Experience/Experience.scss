@import 'styles/utils.scss';

.experience {
  grid-area: experience;

  h3 {
    text-align: right;
  }
}

.job {
  text-align: right;
  margin-bottom: $spacing * 2;

  .dates {
    margin-bottom: $spacing / 2;
    font-weight: $font-weight-bold;
    font-size: $font-size-small;
    text-transform: uppercase;
    color: $color-secondary;
  }

  .details {
    grid-area: details;

    h4 {
      font-weight: $font-weight-bold;
      font-size: $font-size-regular-plus;
    }

    h5 {
      font-style: italic;
      font-size: $font-size-small;
    }

    p {
      margin-top: $spacing;
      text-align: justify;
    }

    .bold {
      font-weight: $font-weight-bold;
    }

  }

}
