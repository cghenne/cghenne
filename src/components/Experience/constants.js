export const jobs = [
  {
    start_date: 'January 2018',
    end_date: null,
    position: 'Senior Full-Stack Engineer',
    company: 'LifeWorks',
    location: 'Toronto, CA',
    description: `LifeWorks transfered me to their Toronto office to direct the
      development support in the Eastern Time Zone, continue the evolution and
      innovation of the product and create opportunities to build out a North
      America-based development team.`,
    tech_stack: null
  },
  {
    start_date: 'July 2016',
    end_date: 'December 2017',
    position: 'Lead Front-End Engineer',
    company: 'LifeWorks',
    location: 'London, UK',
    description: `In addition to front-end development,
      I performed hiring, training, and mentoring
      for the front-end team. I was also involved in project
      development and oversight and have helped to improve the
      operating platform.`,
    tech_stack: null
  },
  {
    start_date: 'July 2015',
    end_date: 'July 2016',
    position: 'Front-End Engineer',
    company: 'LifeWorks',
    location: 'London, UK',
    description: `LifeWorks: A Global Employee Engagement
      Platform serving 50,000 companies with over 15 million users.`,
    tech_stack: `JavaScript, React, AngularJS, HTML, CSS, Sass,
      Redux, Flow, Node.js, Webpack, Gulp, NPM, Clojure, ClojureScript,
      Jenkins, Vagrant`
  },
  {
    start_date: 'May 2012',
    end_date: 'July 2015',
    position: 'Senior Web Developer',
    company: 'Vanilla Active',
    location: 'London, UK',
    description: `Vanilla Active is a digital agency offering services to
      agencies and brands. We designed and implemented promotional web
      sites directly with brands and worked closely with advertising
      and marketing agencies.`,
    tech_stack: 'HTML, CSS, JavaScript, AngularJS, ColdFusion, PHP, Wordpress',
    print_margin: true
  },
  {
    start_date: 'November 2011',
    end_date: 'March 2012',
    position: 'Excel Expert',
    company: 'Wild Media Ltd',
    location: 'London, UK',
    description: `Wild Media Ltd is a startup in the new
      media area. As an Excel expert and
      database manager, I was in charge of optimising
      and automating the logistics department.`,
    tech_stack: null
  },
  {
    start_date: 'July 2011',
    end_date: 'November 2011',
    position: 'Consultant',
    company: 'De Valck Consultants by Altran',
    location: 'Brussels, Belgium',
    description: `De Valck Consultants is part of Altran group,
      the European leader in innovation consulting.
      It is the dedicated brand to Quality Assurance
      for the Life Science industry. I worked as an
      IS Validation Engineer at Baxter Healthcare.`,
    tech_stack: null
  },
  {
    start_date: 'December 2008',
    end_date: 'June 2011',
    position: 'Web Developer',
    company: 'Jules Bordet Institute',
    location: 'Brussels, Belgium',
    description: `The BrEAST data centre is a specialised clinical
      trials unit created in order to conduct large,
      international Phase III studies in breast cancer
      aiming to register new drugs. I worked as a web
      and application developer, in IT support and
      maintenance, and in validation of computer systems
      developed in-house (GCP, 21 CRF part 11).`,
    tech_stack: 'HTML, CSS, JavaScript, PHP, Java, Oracle SQL, MySQL'
  }
];

export type JobT = {
  start_date: string,
  end_date: ?string,
  position: string,
  company: string,
  location: string,
  description: ?string,
  tech_stack: ?string,
  print_margin?: boolean
};
