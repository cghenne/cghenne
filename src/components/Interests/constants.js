export const interests = [
  {
    type: 'sports',
    label: 'Sports',
    description: 'Outdoor sports, squash, badminton'
  },
  {
    type: 'Music',
    label: 'Music',
    description: 'Piano, Guitar'
  },
  {
    type: 'other',
    label: 'Other',
    description: 'Travelling'
  }
];

export type InterestT = {
  type: string,
  label: string,
  description: string
};
