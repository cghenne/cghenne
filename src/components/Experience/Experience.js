// @flow
import React from 'react';
import Job from './Job';
import { jobs } from './constants';
import type { JobT } from './constants';
import './Experience.scss';

const Experience = () => {
  return (
    <div className="box experience">
      <h3>Experience</h3>
      {jobs.map((job: JobT, index: number) => (
        <Job {...job} key={index} />
      ))}
    </div>
  );
};

Experience.displayName = 'Experience';

export default Experience;
