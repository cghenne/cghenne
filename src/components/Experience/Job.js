// @flow
import React from 'react';
import type { JobT } from './constants';
import './Experience.css';

type Props = JobT;

const Job = ({
  start_date,
  end_date,
  position,
  company,
  location,
  description,
  tech_stack
}: Props) => {
  return (
    <div className="job">
      <div className="dates">
        {start_date} - {end_date ? end_date : 'Present'}
      </div>
      <div className="details">
        <h4>{position}</h4>
        <h5>{`${company}, ${location}`}</h5>

        {description && <p>{description}</p>}
        {tech_stack && (
          <p>
            <span className="bold">Tech stack: </span>
            {tech_stack}
          </p>
        )}
      </div>
    </div>
  );
};

Job.displayName = 'Job';

export default Job;
