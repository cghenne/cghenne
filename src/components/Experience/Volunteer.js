// @flow
import React from 'react';
import Job from './Job';
import { jobs } from './constants';
import type { JobT } from './constants';
import './Experience.scss';

const Experience = () => {
  return (
    <div className="box volunteer">
      <h3>Volunteering</h3>
      {jobs
        .filter(job => job.volunteer)
        .map((job: JobT, index: number) => (
          <Job {...job} key={index} />
        ))}
    </div>
  );
};

Experience.displayName = 'Experience';

export default Experience;
