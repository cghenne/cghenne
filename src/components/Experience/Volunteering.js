// @flow
import React from 'react';
import Job from './Job';
import { jobs } from './constants';
import type { JobT } from './constants';
import './Experience.scss';

const Volunteering = () => {
  return (
    <div className="box volunteering">
      <h3>Volunteering</h3>
      {jobs
        .filter(job => job.volunteer)
        .map((job: JobT, index: number) => (
          <Job {...job} key={index} />
        ))}
    </div>
  );
};

Volunteering.displayName = 'Volunteering';

export default Volunteering;
