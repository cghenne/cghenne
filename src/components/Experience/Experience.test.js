import React from 'react';
import { shallow } from 'enzyme';
import Experience from './Experience';
import { jobs } from './constants';

describe('Experience', () => {
  const wrapper = shallow(<Experience />);

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render title', () => {
    expect(wrapper.find('h3').text()).toEqual('Experience');
  });

  it('should render the box with a class of box and experience', () => {
    expect(wrapper.find('div').prop('className')).toEqual('box experience');
  });

  it('should render a Job component for each job in the list of jobs', () => {
    expect(wrapper.find('Job').length).toEqual(jobs.length);
  });
});
