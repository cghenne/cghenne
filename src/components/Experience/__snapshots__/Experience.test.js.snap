// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experience should render without crashing 1`] = `
<div
  className="box experience"
>
  <h3>
    Experience
  </h3>
  <Job
    company="LifeWorks"
    description="LifeWorks transfered me to their Toronto Office to direct the
        development support in the Eastern Time Zone, continue the evolution and
        innovation of the product and create opportunities to build out a North
        America-based development team."
    end_date={null}
    key="0"
    location="Toronto, CA"
    position="Senior Full-Stack Engineer"
    start_date="January 2018"
    tech_stack={null}
  />
  <Job
    company="LifeWorks"
    description="In addition to front-end development,
        I performed hiring, training, and mentoring
        for the front-end team. I was also involved in project
        development and oversight and have helped to improve the
        operating platform."
    end_date="December 2017"
    key="1"
    location="London, UK"
    position="Lead Front-End Engineer"
    start_date="July 2016"
    tech_stack={null}
  />
  <Job
    company="LifeWorks"
    description="LifeWorks: A Global Employee Engagement
        Platform serving 50,000 companies with over 15 million users."
    end_date="July 2016"
    key="2"
    location="London, UK"
    position="Front-End Engineer"
    start_date="July 2015"
    tech_stack="JavaScript, React, AngularJS, HTML, CSS, Sass,
        Redux, Flow, Node.js, Webpack, Gulp, NPM, Clojure, ClojureScript,
        Jenkins, Vagrant"
  />
  <Job
    company="Vanilla Active"
    description="Vanilla Active is a digital agency offering services to
        agencies and brands. We designed and implemented promotional web
        sites directly with brands and worked closely with advertising
        and marketing agencies."
    end_date="July 2015"
    key="3"
    location="London, UK"
    position="Senior Web Developer"
    print_margin={true}
    start_date="May 2012"
    tech_stack="HTML, CSS, JavaScript, AngularJS, ColdFusion, PHP, Wordpress"
  />
  <Job
    company="Wild Media Ltd"
    description="Wild Media Ltd is a startup in the new
        media area. As an Excel expert and
        database manager, I was in charge of optimising
        and automating the logistics department."
    end_date="March 2012"
    key="4"
    location="London, UK"
    position="Excel Expert"
    start_date="November 2011"
    tech_stack={null}
  />
  <Job
    company="De Valck Consultants by Altran"
    description="De Valck Consultants is part of Altran group,
        the European leader in innovation consulting.
        It is the dedicated brand to Quality Assurance
        for the Life Science industry. I worked as an
        IS Validation Engineer at Baxter Healthcare."
    end_date="November 2011"
    key="5"
    location="Brussels, Belgium"
    position="Consultant"
    start_date="July 2011"
    tech_stack={null}
  />
  <Job
    company="Jules Bordet Institute"
    description="The BrEAST data centre is a specialised clinical
        trials unit created in order to conduct large,
        international Phase III studies in breast cancer
        aiming to register new drugs. I worked as a web
        and application developer, in IT support and
        maintenance, and in validation of computer systems
        developed in-house (GCP, 21 CRF part 11)."
    end_date="June 2011"
    key="6"
    location="Brussels, Belgium"
    position="Web Developer"
    start_date="December 2008"
    tech_stack="HTML, CSS, JavaScript, PHP, Java, Oracle SQL, MySQL"
  />
</div>
`;
